caddy_test.go

```diff
@@ -61,6 +61,48 @@ func TestIsLoopback(t *testing.T) {
 	}
 }
 
+func TestIsPrivateNetwork(t *testing.T) {
+	for i, test := range []struct {
+		input  string
+		expect bool
+	}{
+		{"10.0.0.0", true},
+		{"10.0.0.5", true},
+		{"10.255.255.255", true},
+		{"10.0.0.5:1234", true},
+		{"172.16.0.0", true},
+		{"172.19.1.1", true},
+		{"172.31.255.255", true},
```

Comment:
Add tests for the edge cases on both sides of the boundary (both `true` and `false` sides).